addons/product/tests/test_variants.py

```diff
@@ -667,7 +667,7 @@ def test_variant_images(self):
         """
         # Pretend setup happened in an older transaction by updating on the SQL layer and making sure it gets reloaded
         # Using _write() instead of write() because write() only allows updating log access fields at boot time
-        before = datetime.now() - timedelta(seconds=1)
+        before = self.env.cr.now() - timedelta(seconds=1)
```

Comment:
I don't see any `cr.now()` in python tests. I'd rather simply increase the `timedelta` to reduce/nullify the problem. 

```suggestion
        before = datetime.now() - timedelta(seconds=30)
```

Reply:
You don't see cr.now() but essentially every `write` goes through it.
increase the timedelta might solve the current problem, but it's not really certain. It still has a gap to cause errors.

Reply:
I know it is not fundamentally correct. But it will be enough, you can even increase the timedelta if you want.

What I meant is that finding literally 0 occurrence of `cr.now()` in the whole test code base rings a bell to me.
There is definitely something I don't know, but that's a red flag for not using it IMHO.
I could try to find a proper answer, but not sure it is worth the time :yum:

Reply:
Well, I just had a triple check with the framework / runbot team and using `cr.now()`, despite not being used a single time in our (test) code is fine.

Let me know if you want to revert back to your initial change or not and I'll r+

Sorry for the wrong assumption :)

Reply:
I have reset

Reply:
Using cr.now() is the best way to get deterministic (and coherent) results. 
For me there is two reason to use fields.datetime.now():
-> if we have no transaction
-> if we need a real time information in a long transaction (usually for logging or processing time analysis)

Most of the time using cr.now() is more accurate since all operations will be considered as the transaction was instant making it easier to compare with write and create date as well as other records modified in the same transaction. This helps to avoid random errors where comparing two date. (`<` vs `<=`)

Reply:
We actually have it in [that same test in master](https://github.com/odoo/odoo/commit/620b0222506a91e73eac6e08f5d17d20e3a23230#diff-c878c8334e9e4bd59048ba173439e786714d19a2c0b780ecb8fe36b4a46c067dR628).
I could not find `cr.now()` in the 14.0 backport - which made me turn it into a `datetime.now()` but then I forgot to restore it during the forward ports. Sorry...